pages/c/_cluster/apps/chart.vue

```diff
@@ -0,0 +1,264 @@
+<script>
+import Loading from '@/components/Loading';
+import ChartMixin from '@/pages/c/_cluster/apps/chart_mixin';
+import Banner from '@/components/Banner';
+import ChartReadme from '@/components/ChartReadme';
+import LazyImage from '@/components/LazyImage';
+import DateFormatter from '@/components/formatter/Date';
+import isEqual from 'lodash/isEqual';
+import { CHART, REPO, REPO_TYPE, VERSION } from '@/config/query-params';
+
+export default {
+  components: {
+    Banner,
+    ChartReadme,
+    DateFormatter,
+    LazyImage,
+    Loading,
+  },
+
+  mixins: [
+    ChartMixin
+  ],
+
+  async fetch() {
+    await this.baseFetch();
+  },
+
+  data() {
+    return {
+      showLastVersions:  10,
+      showMoreVersions:  false,
+    };
+  },
+
+  computed: {
+    versions() {
+      return this.showMoreVersions ? this.mappedVersions : this.mappedVersions.slice(0, this.showLastVersions);
+    },
+
+    clusterToolWarning() {
+      if (!this.existing) {
+        return;
+      }
+      const url = this.$router.resolve(this.appLocation()).href;
+
+      return this.isClusterTool ? this.t('catalog.chart.errors.clusterToolExists', { url }, true) : '';
+    },
+
+    maintainers() {
+      return this.version.maintainers.map((m) => {
+        return {
+          id:   m.name,
+          text: m.name,
+          url:  `mailto:${ m.email }`
+        };
+      });
+    },
+
+  },
+
+  watch: {
+    '$route.query'(neu, old) {
+      if ( !isEqual(neu, old) ) {
+        this.$fetch();
+      }
+    },
+  },
+
+  methods: {
+    install() {
+      this.$router.push({
+        name:   'c-cluster-apps-install',
+        params: {
+          cluster:  this.$route.params.cluster,
+          product:  this.$store.getters['productId'],
+        },
+        query: {
+          [REPO_TYPE]: this.query.repoType,
+          [REPO]:      this.query.repoName,
+          [CHART]:     this.query.chartName,
+          [VERSION]:   this.query.versionName,
+        }
+      });
+    },
+
+  },
+};
+</script>
+
+<template>
+  <Loading v-if="$fetchState.pending" />
+  <div v-else>
+    <div v-if="chart" class="chart-header">
+      <div class="name-logo-install">
+        <div class="name-logo">
+          <h1>
+            <nuxt-link :to="{ name: 'c-cluster-apps-charts' }">
+              {{ t('catalog.chart.header.charts') }}:
+            </nuxt-link>
+            {{ chart.chartDisplayName }} ({{ targetVersion }})
+          </h1>
+          <div v-if="chart.icon" class="logo-container">
+            <div class="logo-bg ml-10">
+              <LazyImage :src="chart.icon" class="logo" />
+            </div>
+          </div>
+        </div>
+        <button v-if="!requires.length && (!clusterToolWarning)" type="button" class="btn role-primary" @click.prevent="install">
```

Comment:
All done. Though upgrade will have 'Upgrade' and edit 'Update', this is the terminology used throughout the rest of the stepper ('update' instead of 'edit' as it's the terminology used with respect to actions)